extensions/parquet/table/src/main/java/io/deephaven/parquet/table/util/S3SeekableChannelProvider.java

```diff
@@ -0,0 +1,87 @@
+package io.deephaven.parquet.table.util;
+
+import io.deephaven.parquet.base.util.SeekableChannelsProvider;
+import io.deephaven.parquet.table.S3ParquetInstructions;
+import org.jetbrains.annotations.NotNull;
+import software.amazon.awssdk.http.async.SdkAsyncHttpClient;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3AsyncClient;
+import software.amazon.awssdk.http.crt.AwsCrtAsyncHttpClient;
+
+import java.io.IOException;
+import java.net.URI;
+import java.nio.channels.SeekableByteChannel;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * {@link SeekableChannelsProvider} implementation that is used to fetch objects from AWS S3 instances.
+ */
+public final class S3SeekableChannelProvider implements SeekableChannelsProvider {
```

Comment:
I don't think this needs to be public (only used internally, could be made package private, moved)?